baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/UnclosedFilesListUsageTests.java

```diff
@@ -0,0 +1,89 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.Before;
+import org.junit.Test;
+
+public final class UnclosedFilesListUsageTests {
+
+    private CompilationTestHelper compilationHelper;
+
+    @Before
+    public void before() {
+        compilationHelper = CompilationTestHelper.newInstance(UnclosedFilesStreamUsage.class, getClass());
+    }
+
+    @Test
+    public void testThrowsOnListWithNoTryWithResources() {
+        compilationHelper
+                .addSourceLines(
+                        "Test.java",
+                        "import java.nio.file.Files;",
+                        "import java.nio.file.Paths;",
+                        "class Test {",
+                        "  void f(String param) {",
+                        "    try {",
+                        "      // BUG: Diagnostic contains: java.nio.file.Files must be called within a try-with",
+                        "      Files.list(Paths.get(\"/tmp\"));",
+                        "    } catch (java.io.IOException e) {",
+                        "    }",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    public void testThrowsOnWalkWithNoTryWithResources() {
+        compilationHelper
+                .addSourceLines(
+                        "Test.java",
+                        "import java.nio.file.Files;",
+                        "import java.nio.file.Paths;",
+                        "class Test {",
+                        "  void f(String param) {",
+                        "    try {",
+                        "      // BUG: Diagnostic contains: java.nio.file.Files must be called within a try-with",
+                        "      Files.walk(Paths.get(\"/tmp\"));",
+                        "    } catch (java.io.IOException e) {",
+                        "    }",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    public void positive() {
+        compilationHelper
+                .addSourceLines(
+                        "Test.java",
+                        "import java.nio.file.Files;",
+                        "import java.nio.file.Path;",
+                        "import java.nio.file.Paths;",
+                        "import java.util.stream.Stream;",
+                        "class Test {",
+                        "  void f(String param) {",
+                        "    try (Stream<Path> files = Files.list(Paths.get(\"/tmp\"))) {",
```

Comment:
how about this case:
```
try (Writer writer = new MyWriter()) {
  Stream<Path> files = Files.list(...);
  files.forEach(writer.write(f));
} catch (IOException e) {}
```

Reply:
The Tree in try's resources cannot be reached from the `Files.list` method invocation, so it still correctly warns.  Added a test case for it too